crates/pallet-domains/src/lib.rs

```diff
@@ -2532,12 +2545,21 @@ impl<T: Config> Pallet<T> {
                 .or(DomainRegistry::<T>::get(domain_id).map(|domain_obj| domain_obj.created_at))
                 .ok_or(BlockTreeError::LastBlockNotFound)?;
 
-        Ok(DomainRuntimeUpgradeAt::<T>::get(runtime_id)
+        Ok(DomainRuntimeUpgradeRecords::<T>::get(runtime_id)
             .into_keys()
             .rev()
             .take_while(|upgraded_at| *upgraded_at > last_block_at)
             .count() as u32)
     }
+
+    // Return the number of domain instance that instantiated with the given runtime
+    fn domain_instance_count(runtime_id: RuntimeId) -> u32 {
+        // TODO: perhaps add another storage to keep track of this number so we don't need
```

Comment:
i agree with this TODO.
I was hoping we add the count to the runtime registry itself and do a migration. 
Not a blocker for this PR but please create an issue for this so we can track

Reply:
Created https://github.com/subspace/subspace/issues/2850